### ISSUE TITLE
Port Data Parallel Training example to multi-backend

### ISSUE THREAD

**Issue by shivance** [keras-team/keras-hub#1177]
Link : https://github.com/keras-team/keras-io/blob/master/examples/nlp/data_parallel_training_with_keras_nlp.py

The tutorial was originally written in Keras, we need to port this to Keras Core with multi backend support.

**Comment by rajchordia14** [comment]
Can I work on this issue?

**Comment by shivance** [comment]
Sure @rajchordia14 ! You are welcome to contribute.
Please open a PR at keras-team/keras-io ! We can use this thread to discuss if you need any help!

**Comment by shivance** [comment]
Hi @rajchordia14 any update? are you stuck somewhere and need help?

**Comment by rajchordia14** [comment]
Working on it, first ever contribution might take time. Apologies for delay

**Comment by pranavvp16** [comment]
@rajchordia14 are you still working on this or can I take over and start working on it

**Comment by shivance** [comment]
@pranavvp16 you can start working on this if you still are interested. Please try to come up with a colab. Then we could iterate on it.

**Comment by pranavvp16** [comment]
fine I'll start working today

**Comment by pranavvp16** [comment]
@shivance I changed the imports according to the new keras-core and ran the example, the example run fine with pytroch as backend but when backend is set to tensorflow there's as error. Below is the link to collab notebook
[pytorch as backend](https://colab.research.google.com/drive/1iiqf3ngZwHigQHm4xCfV_F-qOeJdGV4u?usp=sharing)
[tensorflow as backend](https://colab.research.google.com/drive/1DPeLTYZ666toykCp-ivX9NZaUoIcR9si?usp=sharing)
Seems there is something missing in optimizers file in tensorflow backend

**Comment by pranavvp16** [comment]
I should be opening the issue in keras-core repository right ?

**Comment by shivance** [comment]
keras-nlp

**Comment by pranavvp16** [comment]
@shivance can I make the PR for converted code sample as it works with other backends, and once the above issue is fixed it will work with tensroflow too

**Comment by shivance** [comment]
Yes, please go ahead! Please open seperate issues for each bug you face. Keras team tracks all the work through github issues.
